File: Day1p2.py
def convert_spelled_digits(word):
    spelled_digits = {"one": "1", "two": "2", "three": "3", "four": "4", "five": "5", "six": "6", "seven": "7", "eight": "8", "nine": "9"}
    return "".join(spelled_digits.get(char, char) for char in word.lower())

def sum_calibration_values_from_file(file_path):
    total_sum = 0

    with open(file_path, 'r') as file:
        for line in file:
            # Extract all substrings that represent numbers
            numbers = [word for word in line.split() if word.isdigit() or word.isalpha()]

            # Combine the first and last "digits" to form a two-digit number
            for number in numbers:
                converted_number = convert_spelled_digits(number)
                value = int(converted_number[0] + converted_number[-1])
                total_sum += value

    return total_sum

# Replace 'Day1.txt' with the actual path to your file
file_path = 'Day1.txt'

# Calculate the sum of calibration values from the file
result = sum_calibration_values_from_file(file_path)

# Print the result
print("Sum of calibration values:", result)

File: Day-1/Day1p2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int main() {
    string input;
    // Read input from file
    ifstream file("Day1.txt");

    if (file.is_open()) {
        
        int total = 0;
        string temp = "";
        string numbers[10] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight","nine"};

        while (getline(file, input)) {
            
            string firstDigit = "";
            string lastDigit = "";
            string word = "";
            
            
         
            for(int i = 0; i < input.length(); i++) {

                if(isdigit(input[i])) {
                    if(firstDigit.empty()) {
                        firstDigit = input[i];
                    }
                    lastDigit = input[i];
                    word = "";
                }

                else {
                    word += input[i];
                    for (int j = 0; j < size(numbers); j++) {
                        if(word.find(numbers[j]) != string::npos) {
                            if(firstDigit.empty()) {
                                firstDigit = to_string(j);
                            }
                            lastDigit = to_string(j);
                            word = word.substr(word.length()-2,word.length()-1);
                        }
                    } 
                }
                
            }
            cout << "First digit is " << firstDigit << " and last digit is " << lastDigit << "\n";
            total += stoi(firstDigit+lastDigit);
            cout << "Total is " << total << "\n";

        }

    file.close();
    cout << total;

    }
    return 0;

}